### ISSUE TITLE
Field displayed in TextInput is not equal to field value

### ISSUE THREAD

**Issue by timqha** [facebook/react-native#18334]
When I set a value in the TextInput component, changing it through setState, changes over the text inside onChange are not displayed on the TextInput field value. 

### Environment

Environment:
```
  OS: macOS Sierra 10.12.6
  Node: 8.1.0
  Yarn: 0.24.6
  npm: 5.6.0
  Watchman: 4.7.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 2.3 AI-162.4069837
Packages: (wanted => installed)
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.1
  react-native: 0.54.1 => 0.54.1
```

### Expected Behavior

Field displayed in TextInput is equal to field value

### Actual Behavior

Field displayed in TextInput is not equal to field value

### Steps to Reproduce

react-native init bugInput

```
export default class App extends Component<Props> {

  state = {amount: ''};

  render() {
    return (
      <View style={styles.container}>
        
        <TextInput 
          style={styles.inputs}
          onChangeText={(text) => {
            if (text.length < 4) {
              this.setState({amount: text});
            }

          }}
          placeholder={'0'}
          value={this.state.amount}
        />

        <Text>
        { this.state.amount}
        </Text>
        
      </View>
    );
  }
}
```

After that write more than four symbols.

**Comment by credifiable** [comment]
I am facing a similar issue. @timqha - Did you find any update on this?

**Comment by timqha** [comment]
@credifiable  I rolled back to version 0.53.3

**Comment by mydesweb** [comment]
this is a bug and it is still not fixed on react native 55.3
